### DISCUSSION TITLE
Is it possible to create a database user that is a auth.user?

### DISCUSSION THREAD

**Discussion by KirioXX** [supabase/supabase#3505] [Questions] [answered]
I try to find a way to enable lo-code/no-code tools to talk to my Postgres database with multiple parties and RLS.
To make it possible to use the RLS for the database user (that I have to create for most of these tools)
would it be possible to create a supabase user for a database user and limit this db user to the same rights a supabase user has?

Thank you!

**Comment by soedirgo** [comment, marked as answer]
Hey @KirioXX, you can grant the `postgres` role to your custom role:
```sql
CREATE ROLE my_role INHERIT IN ROLE postgres;
```

**Reply by KirioXX** [reply]
Hey @soedirgo, thanks for the quick response!
When I give my database user the postgres role wouldn't that just mean the user gets full database access?
I was more looking for a way to have the same limitations on a database user that a superbase user has and maybe even tye that user to a superbase user, or do I miss something?

**Reply by soedirgo** [reply]
I see what you mean - you could inherit `authenticated` instead of `postgres`, but then `auth.uid()` would return null so that's no use.

You can instead try the approach here: https://github.com/supabase/supabase/discussions/3525#discussioncomment-1478137, you can create a function that accepts a `text` query and performs that query, then you use that with `.rpc()`.

Keep in mind though there are security implications of doing this, as noted in that discussion.

**Reply by steve-chavez** [reply]
@KirioXX For your case, I think having a `CREATE ROLE no_code_tool LOGIN;` and doing RLS policies with `current_user`(instead of `auth.uid()`) would be better. Then you can use this `no_code_tool` role for a direct connection to the database.

> I was more looking for a way to have the same limitations on a database user that a superbase user has and maybe even tye that user to a superbase user

IIRC, if you use the dashboard to create your policies these will be created for PUBLIC("all roles") so this `no_code_tool` might already inherit some of these policies. It's likely that you'll need to tune them though(make them consider `current_user` for example).

> You can instead try the approach here: #3525 (reply in thread), you can create a function that accepts a text query and performs that query, then you use that with .rpc().

Hm, that will not work assuming the no-code tool requires a db connection.
